concepts/concepts_service.go

```diff
@@ -373,6 +379,74 @@ func (s *ConceptService) validateObject(aggConcept ontology.NewAggregatedConcept
 	return requestError{err.Error()}
 }
 
+func (s *ConceptService) Delete(uuid string, transID string) error {
+	logEntry := s.log.WithUUID(uuid).WithTransactionID(transID)
+
+	query, result := readConceptRelations(uuid)
+	err := s.driver.Read(query)
+	if errors.Is(err, cmneo4j.ErrNoResultsFound) {
+		return ErrNotFound
+	}
+	if err != nil {
+		logEntry.WithError(err).Error("could not find concept to delete")
+		return err
+	}
+
+	// One of the source concepts has incoming relationships
+	if result.Incoming > 0 {
+		return ErrDeleteRelated
+	}
+
+	// Trying to delete a source concept not a canonical.
+	if result.UUID != result.PrefUUID {
+		return ErrDeleteSource
+	}
+
+	// Delete the canonical and all source concepts
+	query = &cmneo4j.Query{
+		Cypher: `
+			MATCH (canonical:Concept{prefUUID:$uuid})<-[r:EQUIVALENT_TO]-(concept:Concept)
+			DELETE r, concept, canonical`,
```

Comment:
The delete will fail if the concept has any outgoing relationships.

```suggestion
			DETACH DELETE concept, canonical`,
```

Reply:
Addressed this and made a test with a concept with outgoing relationships.